scripts/sudo.ps1

```diff
@@ -3,85 +3,86 @@
 
 # open question - Should -NoProfile be used when invoking PowerShell
 BEGIN {
-	if ($__SUDO_TEST -ne $true) {
-		$SUDOEXE = "sudo.exe"
-	}
-	else {
-		if ($null -eq $SUDOEXE) {
-			throw "variable SUDOEXE has not been set for testing"
-		}
-	}
+    if ($__SUDO_TEST -ne $true) {
+        [System.String]$SUDOEXE = 'sudo.exe'
```

Comment:
Why does the assignment need to be prefixed with the type if the right-hand side is already a string anyway?

Reply:
Because it's best practice to explicitly define variable types.

For example, let's compare this

```powershell
$SomeVariable = 'String'
$SomeVariable.GetType().Name
$SomeVariable = $true
$SomeVariable.GetType().Name
```

with this

```powershell
[System.String]$SomeVariable = 'String'
$SomeVariable.GetType().Name
$SomeVariable = $true
$SomeVariable.GetType().Name
```

<br>

When you run them, you will see that the 2nd code doesn't change the type of the variable, but in the first code, the type of the variable can change later again.